### ISSUE TITLE
Migrate to CryptoKit

### ISSUE THREAD

**Issue by maxgoedjen** [maxgoedjen/secretive#234]
https://developer.apple.com/documentation/cryptokit/secureenclave/p256/signing/privatekey/3353165-init

Missed this on 10.15, looks like you can basically use _only_ CryptoKit now and avoid the keychain?

**Comment by KizzyCode** [comment]
Yes you can; it's extremely simple and straight forward compared to the keychain API. I've used it in [one of my projects](https://github.com/KizzyCode/secureenclave-c/tree/master/Sources/SecureEnclave-Swift), and it's basically less than 100 LoC to get signing and ECDH up and running.

However, there also might be some disadvantages:
 - Keys are exported as sealed blob which AFAIK(!) contains no app information, so anyone who has access to this blob can use it. The permissions are sealed within blob, so the SEP will still require authentication if user interactive authentication has been configured during key creation, but it would probably pretty easy to fool the user into signing something malicious.
 - Since the new code cannot be used to access existing secrets, you'd still need to ship and maintain the current keychain-API code for the next years so that users can access their legacy keys. Maybe it would possible to move the keychain code into a an optional legacy plugin?

However I think a combination of CryptoKit and Keychain could be interesting: Create a key and store the resulting blob as application password or similar in the keychain. The OS would ensure that only your app can access the keys by default, however users could manually grant permissions to other apps if appropriate. This could be especially useful for stuff like beta-testing or to eliminate the bus factor.
